dashboard-api/internal/platform/server/handler/studies/create.go

```diff
@@ -21,18 +21,24 @@ type createResponse struct {
 
 // CreateHandler is a gin handler that is used to create
 // a new Study object in the database
-func CreateHandler(repositories storage.Repositories) gin.HandlerFunc {
+func CreateHandler(r storage.Repositories) gin.HandlerFunc {
 	return func(ctx *gin.Context) {
 		var req, err = parseRequest(ctx)
 		if err != nil {
 			ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 			return
 		}
+		orgID, err := r.User.GetUserOrgID(ctx, auth.GetUserIdFrom(ctx))
```

Comment:
Walk me through your logic on this. I would expect this to either be in middleware OR purely in the queries (join users), as we discussed, but this approach seems to me to have the worst of both worlds: we're touching all the queries AND have a separate DB query AND have to put it into every handler manually? Or is the thought to put it into a middleware later as a refactor?

Reply:
Ah right I complely misunderstood our conversation, this is why I was against this approach :sweat_smile: